File: game_plugins/minimal/shared/sc/plugin_minimal/Move.java
package sc.plugin_minimal;

import com.thoughtworks.xstream.annotations.XStreamAlias;

/**
 * ein spielzug
 * 
 * @author sca, tkra
 */
@XStreamAlias(value = "minimal:move")
public final class Move {

	// das betroffene schaf
	public final Sheep sheep;

	// das spielfeld auf das das betroffene schaf ziehen soll
	public final int target;

	public Move(Sheep sheep, int target) {
		this.sheep = sheep;
		this.target = target;
	}

	@Override
	public boolean equals(final Object obj) {
		if (obj instanceof Move) {
			Move other = (Move) obj;
			return other.sheep == sheep && other.target == target;
		}
		return false;
	}

}

File: plugin/src/shared/sc/plugin2021/Move.kt
package sc.plugin2021

import com.thoughtworks.xstream.annotations.XStreamAlias
import sc.api.plugins.IMove

/** Ein abstrakter Spielzug. */
sealed class Move: IMove {
    /** Die [Color], die den Zug getätigt hat. */
    abstract val color: Color
    abstract override fun toString(): String
}

/**
 * Ein Zug, der den gegebenen Spielstein auf dem Spielfeld platziert.
 * @property piece der zu platzierende Spielstein
 */
@XStreamAlias(value = "setmove")
class SetMove(val piece: Piece): Move() {
    override val color: Color get() = piece.color

    override fun toString(): String = piece.toString()
    override fun equals(other: Any?): Boolean = piece == other
    override fun hashCode(): Int = piece.hashCode()
}

/** Ein Zug, der die aktuelle Runde aussetzt. */
@XStreamAlias(value = "skipmove")
class SkipMove(override val color: Color): Move() {
    override fun toString(): String = "$color skipped"
}